src/components/policy/policy_external/test/update_status_manager_test.cc

```diff
@@ -72,18 +73,53 @@ class UpdateStatusManagerTest : public ::testing::Test {
   void TearDown() OVERRIDE {}
 };
 
-// TODO(AKutsan) : APPLINK-31222 use miliseconds as parameter of OnUpdateSentOut
-// function
-TEST_F(
-    UpdateStatusManagerTest,
-    DISABLED_OnUpdateSentOut_WaitForTimeoutExpired_ExpectStatusUpdateNeeded) {
+namespace {
+class WaitAsync {
+ public:
+  WaitAsync(const uint32_t count, const uint32_t timeout)
+      : count_(count), timeout_(timeout), cond_var_() {}
+
+  void Notify() {
+    count_--;
+    cond_var_.NotifyOne();
+  }
+
+  void Wait(sync_primitives::AutoLock& auto_lock) {
+    while (count_ > 0) {
+      sync_primitives::ConditionalVariable::WaitStatus wait_status =
+          cond_var_.WaitFor(auto_lock, timeout_);
+      if (wait_status == sync_primitives::ConditionalVariable::kTimeout) {
```

Comment:
@LitvinenkoIra 
https://en.wikipedia.org/wiki/Spurious_wakeup can occur, so perhaps you need something to prevent that.